include/Parser.h

```diff
@@ -0,0 +1,92 @@
+#ifndef PARSER_H
+#define PARSER_H
+#include <iostream>
+#include <map>
+#include <vector>
+#include <string>
+#include <sstream>
+#include <stdexcept>
+
+class Parser {
+public:
+    Parser(); // Constructor
+
+    bool checkOption(const std::string& name);
+
+    // Add an option with a default value
+    template<typename T>
+    void addOption(const std::string& name, const std::string& description, T defaultValue);
+
+    // Add a flag (boolean option)
+    void addFlag(const std::string& name, const std::string& description);
+
+    // Parse arguments
+    void parse(int argc, const char* argv[]);
+
+    // Parser Setup
+    void parserSetup();
+
+    // Get the value of an option
+    template<typename T>
+    T getOption(const std::string& name) const;
+
+    // Print help/usage information
+    void printHelp() const;
+
+    // List books
+    void listBooks(); 
+
+    // Search for a book
+    void searchBook();
+
+
+private:
+    // Internal Option structure
+    struct Option {
+        std::string description;
+        std::string value;
+        bool inCLI = false;
+        bool isFlag;
+
+        // Default constructor
+        Option() : description(""), value(""), isFlag(false) {}
+
+        // Parameterized constructor
+        Option(const std::string& desc, const std::string& val, bool flag = false)
+            : description(desc), value(val), isFlag(flag) {}
+    };
```

Comment:
put it in its own class unless there's a reason to have nestled in ehre

Reply:
will make in separate pr